### PR TITLE
Configurable storage

### PR DESCRIPTION
**Pull request by gz**
Is this a user-visible change (yes/no): yes

### CODE REVIEW

File: crates/pipeline-types/src/config.rs [hunk 1]
```diff
@@ -83,6 +72,18 @@ pub struct RuntimeConfig {
...
+    ///
+    /// - If `true`, the pipeline state is stored in the specified location,
+    ///   is persisted across restarts, and can be checkpointed and recovered.
+    ///
+    /// - If `false`, the pipeline's state is kept in in-memory data-structures.
+    ///   This is useful if the pipeline is ephemeral and does not need to be recovered
+    ///   after a restart. The pipeline will most likely run faster since it does not
+    ///   need to read from, or write to disk
+    #[serde(default)]
+    pub storage: bool,
```
Comment (lalithsuresh) [inline crates/pipeline-types/src/config.rs:85]: False is the default right? Worth specifying for "true" that this is a preview/experimental feature..

**Review by blp — APPROVED** [review]

File: openapi.json [hunk 2]
```diff
@@ -3407,6 +3407,29 @@
       "CsvParserConfig": {
         "type": "object"
       },
+      "DeltaTableWriterConfig": {
```
[the remark this answers is not in the document]
Comment (gz) [reply openapi.json:3410]: yea it didn't get udpated

[the remark this answers is not in the document]
Comment (lalithsuresh) [reply openapi.json:3410]: I assume it was one of those cases where the CI gave a green check mark even though the run failed. We have a Earthly case stage to catch this.

[the remark this answers is not in the document]
Comment (gz) [reply openapi.json:3410]: the json was up to date but the typescript bindings in the web-console are not necessarily in-sync (and this part isnt checked)

[the remark this answers is not in the document]
Comment (gz) [reply openapi.json:3410]: looks like it, yes

**Pull request merged**